### PR TITLE
DAOS-1122 tests: run all subtests in io and rebuild suites

### PR DESCRIPTION
**Pull request by bfaccini**
This patch allows running all selected sub-tests with io and
rebuild suites, by using cmocka tests group methods which are
the only to continue upon intermediate skip/failure.

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

### CODE REVIEW

**Comment by bfaccini** [comment]
> Avoid using I/O strings. This will break, XML file creation. Some information available under: 
> https://wiki.hpdd.intel.com/pages/viewpage.action?pageId=105317830
eh eh, thanks to highlight this Ravindran !!
To be honest, since I had to modify my original patch during rebase due to the same change with previous "I/O" string in same routine, I have decided to keep this/mine and see how it goes !!...
So, ok I will change this occurence too with only "IO", and now thanks to you I know why !!

File: src/tests/suite/daos_rebuild.c [hunk 1]
```diff
@@ -1899,38 +1919,48 @@ static const struct CMUnitTest rebuild_tests[] = {
 	{"REBUILD23: disconnect pool during scan",
 	 rebuild_tgt_pool_disconnect_in_scan, NULL, test_case_teardown},
 	{"REBUILD24: disconnect pool during rebuild",
-	 rebuild_tgt_pool_disconnect_in_rebuild, NULL, test_teardown},
+	 rebuild_tgt_pool_disconnect_in_rebuild, NULL, test_case_teardown},
```
Comment (wangdi1) [inline src/tests/suite/daos_rebuild.c]: oh, this seems to be changed to rebuild_sub_teardown,  i.e. it needs to cleanup the pool.

File: src/tests/suite/daos_rebuild.c [hunk 2]
```diff
@@ -1846,10 +1848,28 @@ multi_pools_rebuild_concurrently(void **state)
 static int
 rebuild_sub_setup(void **state)
 {
+	if (state != NULL && *state != NULL) {
+		save_arg = *state;
+		*state = NULL;
+	}
```
Comment (wangdi1) [inline src/tests/suite/daos_rebuild.c]: If we can re-order these subtest case to make all cases, which share the same states, are stick together, do we still need this?

Comment (bfaccini) [reply src/tests/suite/daos_rebuild.c]: no but this is intended to safely handles all cases...

File: src/tests/suite/daos_rebuild.c (hunk 1 above)
Comment (bfaccini) [reply src/tests/suite/daos_rebuild.c]: Why ? My understanding is that the current/main/group opened pool must not be cleaned between sub-tests, only the ones that are used per sub-test and created by rebuild_sub_stup() if used.

**Comment by bfaccini** [comment]
Well, looks like all tests have been successful, only defect is for "DCO" step and due to a missing "Signed-off-by" in one of my old push/commit, so can you review this patch again so it can land ??!!

**Comment by bfaccini** [comment]
Can I get a review or must I forget about this patch forever  ?? ;-)

**Pull request merged**